Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1252,16 +1252,16 @@ void MediaPlayerPrivateGStreamer::videoSinkCapsChanged(GstPad* videoSinkPad)
     });
 }
 
-void MediaPlayerPrivateGStreamer::handleTextSample(GstSample* sample, const char* streamId)
+void MediaPlayerPrivateGStreamer::handleTextSample(GRefPtr<GstSample>&& sample, StringView streamId)
```

Comment:
I wonder why you keep here a StringView, which also not owns the internal string, if the caller seems to be the only place using a String. Is there another rework ongoing where having here a StringView makes sense? If it does not it would make sense to change this into a `const String&`.

And even if there is another rework ongoing and you wish to keep this as a StringView, you can switch it to `const StringView&`.

Reply:
I don't see any difference between using String or StringView here because the caller owns the string, but ok, whatever :)

Reply:
I meant there was no difference in using StringView here from the memory handling side but it does not make sense from the architectural side since it was going to come from a String (unless you have some other rework pending I am missing). In this case, turning it into const String& was better. And if you had some rework pending with StringView, you could still turn it into a `const StringView&`. That was my point, I hope I manage to explain it properly.

Actually, we should have done the `streamId` parameter inside the priv a String already, please keep it in mind if you do further rework around here.